### PR TITLE
prometheues: add kiwi template for SLE_15_SP1 and OpenSUSE tumbleweed

### PR DESCRIPTION
**Pull request by sjamgade**
builds at https://build.suse.de/package/show/home:sjamgade:Containers/golang-github-prometheus-prometheus-image

### CODE REVIEW

File: golang-github-prometheus-prometheus-image/golang-github-prometheus-prometheus-image.kiwi.j2 [hunk 1]
```diff
@@ -0,0 +1,46 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  This file is generated, don't manually modify this file!
+-->
+<image name="golang-github-prometheus-prometheus-image" schemaversion="6.9" xmlns:suse_label_helper="com.suse.label_helper">
```
Comment (witekest) [inline golang-github-prometheus-prometheus-image/golang-github-prometheus-prometheus-image.kiwi.j2:5]: Do we have some CI relying on the package name _golang-github-prometheus-prometheus_ and do we have to use it for directory name as well? _prometheus-image_ looks much nicer.

File: golang-github-prometheus-prometheus-image/golang-github-prometheus-prometheus-image.kiwi.j2 [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  <preferences>
+    <type image="docker" derived_from="{{ derived_from }}">
+      <containerconfig user="prometheus" additionaltags="%LONG_VERSION%,%LONG_VERSION%-%RELEASE%" maintainer="SUSE Cloud Team &lt;cloud@suse.com&gt;" name="prometheus">
+        <expose>
+          <port number="9090"/>
+        </expose>
+        <entrypoint execute="/usr/bin/prometheus"/>
+        <subcommand clear="true"/>
+        <labels>
+          <label name="org.opencontainers.image.title" value="Prometheus containe"/>
```
Comment (witekest) [inline golang-github-prometheus-prometheus-image/golang-github-prometheus-prometheus-image.kiwi.j2:26]: Typo: Prometheus container

**Review by witekest — COMMENTED** [review]
I think we should define a Docker volume to use for storing measurements.

File: golang-github-prometheus-prometheus-image/golang-github-prometheus-prometheus-image.kiwi.j2 (hunk 1 above)
Comment (sjamgade) [reply golang-github-prometheus-prometheus-image/golang-github-prometheus-prometheus-image.kiwi.j2:5]: currently I did find any convention, so went ahead with package name. 
But I agree that `prometheus-image` is better

**Comment by sjamgade** [comment]
> I think we should define a Docker volume to use for storing measurements.

we could, but that iiuc is defined either by the helm chart or the manifest deploying this image. So that storage could be mounted when the image is being deployed. Isnt that so ?

**Comment by witekest** [comment]
> > I think we should define a Docker volume to use for storing measurements.
> 
> we could, but that iiuc is defined either by the helm chart or the manifest deploying this image. So that storage could be mounted when the image is being deployed. Isnt that so ?

The image should be able to run with Docker client only. Running `docker run` should create a new volume and mount it at the defined mount point. Please compare [[1](https://docs.docker.com/engine/reference/builder/#volume), [2](https://github.com/prometheus/prometheus/blob/master/Dockerfile)].

Helm chart will include additional information about Kubernetes volume.

**Comment by sjamgade** [comment]
added volume info
create volume `docker volume create --opt o=uid=483 --opt device=tmpfs --opt type=tmpfs drv`
run image : ` docker run  --mount 'type=volume,src=drv,destination=/prometheus' -p 9090:9090 62a05d1a2a65`

```
╰─ docker volume inspect drv
[
    {
        "CreatedAt": "2019-10-02T18:58:14+02:00",
        "Driver": "local",
        "Labels": {},
        "Mountpoint": "/var/lib/docker/volumes/drv/_data",
        "Name": "drv",
        "Options": {
            "device": "tmpfs",
            "o": "uid=483",
            "type": "tmpfs"
        },
        "Scope": "local"
    }
]
```

**Comment by JanZerebecki** [comment]
Will not be used any more.

**Pull request closed without merging**